107.py

```diff
@@ -0,0 +1,39 @@
+from typing import List
+from collections import deque
+
+
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:
+        levels = []
+
+        if not root:
+            return levels
+        
+        queue = deque([root])
+        
+        while queue:
+            level = []
+            level_length = len(queue)
+            
+            while level_length > 0:
```

Comment:
This BFS implementation is very close to the BFS template I always use. However, instead of maintaining `level_length` yourself, why don't we just use a `for` loop, especially when `level_length` is a variable that has to be decremented every time regardless of what happens in the loop? This is a good opportunity to think about which scenarios are better suited for a `while` loop and which are for a `for` loop.